lib/serializer/src/types.js

```diff
@@ -319,10 +319,10 @@ Types.time_point_sec = {
             return Math.floor( object.getTime() / 1000 );
 
         if(typeof object !== "string")
-            throw new Error("Unknown date type: " + object)
+            throw new Error("Unknown date type: " + object);
 
-        // if(typeof object === "string" && !/Z$/.test(object))
-        //     object = object + "Z"
+        if(/T[0-2][0-9]:[0-5][0-9]:[0-5][0-9]$/.test(object))
```

Comment:
I'm ok with restoring the "Z" addition here, but I'm not sure your regex is a good idea. If the serialization is changed in the backend to actually include the Z this regex will cause us to add an additional Z. Why do you feel the original !Z check is insufficient?

Reply:
the regex has a $ in the end, which makes a real Z at the end will not work. 

/T[0-2][0-9]:[0-5][0-9]:[0-5][0-9]$/.test('2018-02-06T20:13:25Z') => false
/T[0-2][0-9]:[0-5][0-9]:[0-5][0-9]$/.test('2018-02-06T20:13:25') => true

a date/time string may have many different formats, i.e. 

  2018-10-01 05:03:03+0800

in many cases just adding Z in the end is wrong.

Reply:
Ok, thanks for the explanation!